tests/integrations/client/client_test.go

```diff
@@ -2473,6 +2499,119 @@ func (s *clientStatefulTestSuite) TestGCBarriers() {
 	}
 }
 
+func (s *clientStatefulTestSuite) TestGlobalGCBarriers() {
+	s.prepareKeyspacesForGCTest()
+	re := s.Require()
+	ctx := context.Background()
+
+	keyspaceID := constants.NullKeyspaceID
+	cli := s.client.GetGCStatesClient(keyspaceID)
```

Comment:
Again, you covered only the null keyspace.
What you should consider is:
* Global GC barrier APIs works the same way when called on clients of different keyspace (at least for now; we may need to consider permission control in the future).
* Global GC barriers affect and are affected by all keysapces
* It should work as normal when null keyspace is never used (which is the next-gen scenario), when only null keyspace is used (which is the classic scenario), or mixed usage (which is the serverless scenario).